hs_core/hydro_realtime_signal_processor.py

```diff
@@ -5,10 +5,50 @@
 import types
 from haystack.query import SearchQuerySet
 from haystack.utils import get_identifier
+from hs_core.models import SOLRQueue
 
 logger = logging.getLogger(__name__)
 
 
+def solr_update(instance): 
+    """ Update a resource's SOLR record """
+    from hs_core.models import BaseResource
+    from hs_core.search_indexes import BaseResourceIndex
+    logger.info('updating {}'.format(instance.short_id))
+    index = BaseResourceIndex()
+    newbase = BaseResource.objects.get(pk=instance.pk)
+    try:
+        index.update_object(newbase)
+    except NotHandled:
+        logger.exception("Failure: changes to %s with short_id %s not added to Solr Index.",
+                         str(type(instance)), newbase.short_id)
+
+
+def solr_delete(instance): 
+    """ Delete a resource from SOLR before deleting from Django """
+    from hs_core.models import BaseResource
+    from hs_core.search_indexes import BaseResourceIndex
+    logger.info('deleting {}'.format(instance.short_id))
+    index = BaseResourceIndex()
+    newbase = BaseResource.objects.get(pk=instance.pk)
+    try:
+        index.remove_object(newbase)
+    except NotHandled:
+        logger.exception("Failure: delete of %s with short_id %s failed.",
+                         str(type(instance)), newbase.short_id)
+
+
+def solr_batch_update(): 
+    """ update SOLR for resources in the SOLRQueue """
+    logger.info('starting batch update') 
+    for instance in SOLRQueue.read_and_clear(): 
+        logger.info('checking {}'.format(instance.short_id))
+        if instance.show_in_discover:  # if object should be displayed now
+            solr_update(instance)
+        else:  # not to be shown in discover
+            solr_delete(instance)
```

Comment:
Because a resource that is deleted will cause an error when trying to receive the resource, resources that come after the deleted resource will not be processed.